File: flutter_app/mrs_unkwn_app/lib/core/permissions/family_permissions.dart
import "../../features/family/data/models/family.dart";

enum FamilyPermission { manageMembers, viewActivity, editSettings, deleteFamily }

class FamilyPermissions {
  const FamilyPermissions._();

  static const Map<FamilyRole, Set<FamilyPermission>> _rolePermissions = {
    FamilyRole.parent: {
      FamilyPermission.manageMembers,
      FamilyPermission.viewActivity,
      FamilyPermission.editSettings,
      FamilyPermission.deleteFamily,
    },
    FamilyRole.admin: {
      FamilyPermission.manageMembers,
      FamilyPermission.viewActivity,
      FamilyPermission.editSettings,
      FamilyPermission.deleteFamily,
    },
    FamilyRole.guardian: {
      FamilyPermission.manageMembers,
      FamilyPermission.viewActivity,
    },
    FamilyRole.child: {
      FamilyPermission.viewActivity,
    },
  };

  static bool hasPermission(
    FamilyRole role,
    FamilyPermission permission,
  ) {
    return _rolePermissions[role]?.contains(permission) ?? false;
  }
}

File: backend/src/utils/familyPermissions.ts
import { User } from '../repositories/user.repository';

export type FamilyPermission =
  | 'MANAGE_MEMBERS'
  | 'VIEW_ACTIVITY'
  | 'EDIT_SETTINGS'
  | 'DELETE_FAMILY';

export const rolePermissions: Record<User['role'], FamilyPermission[]> = {
  parent: ['MANAGE_MEMBERS', 'VIEW_ACTIVITY', 'EDIT_SETTINGS', 'DELETE_FAMILY'],
  child: ['VIEW_ACTIVITY'],
};
